[libc++][Android] Explicitly declare low-level lib existence

rprichard opened the pull request:
Android's librt and libpthread functionality is part of libc.{a,so} instead. The atomic APIs are part of the compiler-rt builtins archive. Android does have libdl.

Android's libc.so has `__cxa_thread_atexit_impl` starting in API 23, and the oldest supported API is 21, so continue using feature detection for that API.

These settings need to be declared explicitly for the sake of the fuzzer library's custom libc++ build `add_custom_libcxx`. That macro builds libc++ using `-DCMAKE_TRY_COMPILE_TARGET_TYPE=STATIC_LIBRARY`, which breaks the feature detection.

ldionne reviewed (approved):
LGTM but please update on top of `main` (or merge `main` into your PR branch) to get a green CI run.

ldionne commented:
Then you can squash and merge yourself.

rprichard committed:
[libc++][Android] Explicitly declare low-level lib existence

Android's librt and libpthread functionality is part of libc.{a,so}
instead. The atomic APIs are part of the compiler-rt builtins archive.
Android does have libdl.

Android's libc.so has `__cxa_thread_atexit_impl` starting in API 23,
and the oldest supported API is 21, so continue using feature detection
for that API.

These settings need to be declared explicitly for the sake of the
fuzzer library's custom libc++ build `add_custom_libcxx`. That macro
builds libc++ using `-DCMAKE_TRY_COMPILE_TARGET_TYPE=STATIC_LIBRARY`,
which breaks the feature detection.
